### PR TITLE
Added files

### PR DESCRIPTION
**Pull request by michallibov**
Fixed the previous pull request (of the deleted files)

### CODE REVIEW

File: cup-of-join.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    """
+    :param lists: the lists we got from the main to join
+    :param sep: the char that will separate each list
+    :return: the joined list of all lists the function received
+    """
+    if not lists:   # the list we got is empty (no list was sent)
+        return None
+
+    new_list = list()
+    for x in lists:
```
Comment (talbarfm) [inline cup-of-join.py]: Use list comprehension and zip to remove this forloop

File: cup-of-join.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+        new_list.extend(x)      # adds the given list to the output list without []
```
Comment (talbarfm) [inline cup-of-join.py]: Single line comments start with capital letter and on a separate line

File: cup-of-join.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+# list where each list we got is separated by a character the user chose.
+# if there's no separator- the default separator is going to be '-'.
+def join(*lists, sep='-'):
+    """
+    :param lists: the lists we got from the main to join
+    :param sep: the char that will separate each list
+    :return: the joined list of all lists the function received
+    """
+    if not lists:   # the list we got is empty (no list was sent)
+        return None
```
Comment (talbarfm) [inline cup-of-join.py]: https://stackoverflow.com/questions/1839289/why-should-functions-always-return-the-same-type

File: piece-of-cake.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
+"""
```
Comment (talbarfm) [inline piece-of-cake.py]: You dont need this documentation, the function documentation is enough

File: piece-of-cake.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+# we need, and calculates the total price of the ingredients we must buy.
+def get_recipe_price(prices, optionals=[], **ingredients):
+    """
+    :param prices: a dictionary of the names and prices of an ingredient
+    :param optionals: a list of optional ingredients we don't have to buy
+    :param ingredients: the amount of each ingredient we need
+    :return: the total price we will have to pay
+    """
+    total_price = 0
+    for ingredient in ingredients:
```
Comment (talbarfm) [inline piece-of-cake.py]: use functools.reduce to remove this for loop

File: piece-of-cake.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        if ingredient not in optionals:     # if we must buy this ingredient (doesn't appear in optional)
+            total_price += prices.get(ingredient) * ingredients.get(ingredient) // 100
+    return total_price
+
+
+print(get_recipe_price({'chocolate': 18, 'milk': 8}, milk=100, chocolate=200))
```
Comment (talbarfm) [inline piece-of-cake.py]: Add newline at eof

Comment (talbarfm) [inline piece-of-cake.py]: surround this in name == main 
read this to understand  https://www.freecodecamp.org/news/if-name-main-python-example/

File: cup-of-join.py [hunk 7]
```diff
@@ -0,0 +1,30 @@
+"""
+In this program I rewrote the known function 'join' for lists. This function gets
```
Comment (talbarfm) [inline cup-of-join.py:2]: you don't need this description

**Review by talbarfm — CHANGES_REQUESTED** [review]

File: cup-of-join.py (hunk 1 above)
Comment (michallibov) [reply cup-of-join.py]: I was trying to figure out how to do that in a better way than how I did but couldn't find a way to get rid of the parenthesis inside the list. I also didn't find a way to not use the pop function after the list comprehension. I know it's not perfect..

Comment (michallibov) [reply cup-of-join.py]: I managed to succeed doing it and now it works fine:P

**Review by lovodkin93 — APPROVED** [review]
well done!

File: piece-of-cake.py [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+    This function gets ingredients, their prices, optional ingredients and how much of them
+    we need, and calculates the total price of the ingredients we must buy.
+    :param prices: a dictionary of the names and prices of an ingredient
+    :param optionals: a list of optional ingredients we don't have to buy
+    :param ingredients: the amount of each ingredient we need
+    :return: the total price we will have to pay
+    """
+
+    # Sums up the total price of all ingredients we must buy using a lambda function.
+    total_price = functools.reduce(lambda price, ingredient: price + ingredients.get(ingredient) *
```
Comment (itayday) [inline piece-of-cake.py:15]: Great!

**Pull request merged**